internal/model/model.go

```diff
@@ -57,14 +61,24 @@ type Asset struct {
 	BMCAddress net.IP
 }
 
-// IncludeError includes the given error key and value in the asset
+// AppendError includes the given error key and value in the asset
 // which is then available to the publisher for reporting.
-func (a *Asset) IncludeError(key, value string) {
+func (a *Asset) AppendError(key CollectorError, value string) {
 	if a.Errors == nil {
 		a.Errors = map[string]string{}
 	}
 
-	a.Errors[key] = value
+	a.Errors[string(key)] = value
+}
+
+func (a *Asset) HasError(cErr CollectorError) bool {
+	for k := range a.Errors {
```

Comment:
Why not do a map lookup here? Am I missing something?

Reply:
:slaps forehead, yeah this was me being stupid

Reply:
fixed in https://github.com/metal-toolbox/alloy/pull/146/commits/a5bcee07d59eb6616cf14353fffccedd6caed746